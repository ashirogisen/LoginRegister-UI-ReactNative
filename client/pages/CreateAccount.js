import React, { useState } from 'react'
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native'

import Form from '../components/Form'
import Button from '../components/Button'

function CreateAccount(props) {
  const [ name, setName ] = useState('')
  const [ email, setEmail ] = useState('')
  const [ password, setPassword ] = useState('')
  const [ repeatPassword, setRepeatPassword] = useState('')

  const backToSignIn = () => {
    props.navigation.navigate('Login')
  }

  const TestFunction = () => {
    console.log(name)
    console.log(email)
    console.log(password)
    console.log(repeatPassword)
  }
  
  const styles = StyleSheet.create({
    mainContainer: {
      justifyContent: 'center',
      alignItems: 'center',
      height: '100%',
      backgroundColor: '#555555',
    },
    title: {
      color: 'white',
    },
    formsContainer: {
      width: '90%',
      marginVertical: 20
    },
    buttonContainer: {
      width: '90%',
      marginBottom: 40
    },
    footerContainer: {
      width: '100%',
      justifyContent: 'flex-start',
      alignItems: 'flex-start',
      marginLeft: 50
    },
    footerMenus: {
      color: 'white',
    }
  })

  const buttonCustomStyle = {
    backgroundColor: '#555555',
    borderWidth: 1,
    borderColor: 'white',
    justifyContent: 'center',
    alignItems: 'center',
  }

  const x = "< Sign In"
  const forms = [
    { value: name, setValue: setName, icon: 'md-person', placeholder: 'Name', iconColor: 'white' },
    { value: email, setValue: setEmail, icon: 'md-mail', placeholder: 'Email', iconColor: 'white' },
    { value: password, setValue: setPassword, icon: 'md-lock', placeholder: 'Password', iconColor: 'white' },
    { value: repeatPassword, setValue: setRepeatPassword, icon: 'md-lock', placeholder: 'Repeat Password', iconColor: '#555555' },
  ]

  return(
    <View testID="test-mainContainer" style={styles.mainContainer}>
      <View>
        <Text testID="test-title" style={styles.title}>CREATE ACCOUNT</Text>
      </View>
      <View testID="test-formsContainer" style={styles.formsContainer}>
        {
          forms.map((form, index) => (
            <Form key={index} {...form}  />
          ))
        }
      </View>
      <View testID="test-buttonContainer" style={styles.buttonContainer}>
        <Button command={TestFunction} customStyles={buttonCustomStyle} name="Continue"/>
      </View>
      <View testID="test-footerContainer" style={styles.footerContainer}>
        <TouchableOpacity onPress={backToSignIn}>
          <Text testID="test-footerMenu" style={styles.footerMenus}>{x}</Text>
        </TouchableOpacity>
      </View>
    </View>
  )

}

export default CreateAccount