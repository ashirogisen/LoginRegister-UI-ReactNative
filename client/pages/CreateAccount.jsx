import React from 'react'
import { StyleSheet, View, Text } from 'react-native'

import Form from '../components/Form'
import Button from '../components/Button'

function CreateAccount() {
  const styles = StyleSheet.create({
    container: {

    },
    title: {

    },
    formsContainer: {

    },
    buttonContainer: {

    },
    footerContainer: {

    }
  })

  return(
    <View>
      <View style={styles.container}>
        <Text>CREATE ACCOUNT</Text>
      </View>
      <View style={styles.formsContainer}>
        <Form icon="md-person" placeholder="Name" />
        <Form icon="md-mail" placeholder="Email" />
        <Form icon="md-lock" placeholder="Password"/>
        <Form icon="md-lock" placeholder="Repeat Password" />
      </View>
      <View>
        <Button name="Continue"/>
      </View>
      <View>
        <Text>Sign In</Text>
      </View>
    </View>
  )

}

export default CreateAccount