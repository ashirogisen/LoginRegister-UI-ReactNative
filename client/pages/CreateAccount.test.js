import React from 'react'
import CreateAccount from './CreateAccount'
import renderer from 'react-test-renderer'
import { render } from 'react-native-testing-library'

describe('Render CreateAccount', () => {
  it('Renders without crashing', () => {
    const tree = renderer.create(
      <CreateAccount />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  })

  it('Should render main container', () => {
    const { debug, getByTestId } = render (
      <CreateAccount />
    )
    // debug()
    expect(getByTestId('test-mainContainer')).toBeTruthy()
  })

  it('Should render title', () => {
    const { getByTestId } = render (
      <CreateAccount />
    )
    expect(getByTestId('test-title')).toBeTruthy()
  })

  it('Should render forms container', () => {
    const { getByTestId } = render (
      <CreateAccount />
    )
    expect(getByTestId('test-formsContainer')).toBeTruthy()
  })

  it('Should render buttons container', () => {
    const { getByTestId } = render (
      <CreateAccount />
    )
    expect(getByTestId('test-buttonContainer')).toBeTruthy()
  })

  it('Should render footers container', () => {
    const { getByTestId } = render (
      <CreateAccount />
    )
    expect(getByTestId('test-footerContainer')).toBeTruthy()
  })

  it('Should render footers menu', () => {
    const { getByTestId } = render (
      <CreateAccount />
    )
    expect(getByTestId('test-footerMenu')).toBeTruthy()
  })
})
