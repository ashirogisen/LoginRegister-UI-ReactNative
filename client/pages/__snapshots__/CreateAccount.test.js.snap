// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CreateAccount Renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#555555",
      "height": "100%",
      "justifyContent": "center",
    }
  }
  testID="test-mainContainer"
>
  <View>
    <Text
      style={
        Object {
          "color": "white",
        }
      }
      testID="test-title"
    >
      CREATE ACCOUNT
    </Text>
  </View>
  <View
    style={
      Object {
        "marginVertical": 20,
        "width": "90%",
      }
    }
    testID="test-formsContainer"
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#888888",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 1,
          "borderRadius": 30,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "color": "white",
          "flexDirection": "row",
          "margin": 10,
          "marginLeft": 2,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="test-formItem"
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Name"
        placeholderTextColor="#bdc3c7"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 8,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        testID="test-formInput"
        underlineColorAndroid="rgba(0,0,0,0)"
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#888888",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 1,
          "borderRadius": 30,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "color": "white",
          "flexDirection": "row",
          "margin": 10,
          "marginLeft": 2,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="test-formItem"
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#bdc3c7"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 8,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        testID="test-formInput"
        underlineColorAndroid="rgba(0,0,0,0)"
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#888888",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 1,
          "borderRadius": 30,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "color": "white",
          "flexDirection": "row",
          "margin": 10,
          "marginLeft": 2,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="test-formItem"
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#bdc3c7"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 8,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        testID="test-formInput"
        underlineColorAndroid="rgba(0,0,0,0)"
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#888888",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 1,
          "borderRadius": 30,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "color": "white",
          "flexDirection": "row",
          "margin": 10,
          "marginLeft": 2,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="test-formItem"
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Repeat Password"
        placeholderTextColor="#bdc3c7"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 8,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        testID="test-formInput"
        underlineColorAndroid="rgba(0,0,0,0)"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 40,
        "width": "90%",
      }
    }
    testID="test-buttonContainer"
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#555555",
          "borderBottomWidth": null,
          "borderColor": "white",
          "borderLeftWidth": null,
          "borderRadius": 57,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": 1,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
      testID="test-button"
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        testID="test-buttonText"
        uppercase={false}
      >
        Continue
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "flex-start",
        "marginLeft": 50,
        "width": "100%",
      }
    }
    testID="test-footerContainer"
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
        testID="test-footerMenu"
      >
        &lt; Sign In
      </Text>
    </View>
  </View>
</View>
`;
